src/community/script/js/src/main/java/org/geoserver/script/js/JsgiRequest.java

```diff
@@ -36,41 +30,41 @@
      * @param cx
      * @param scope
      */
-    public JsgiRequest(Request request, Response response, Context cx, Scriptable scope) {
+    public JsgiRequest(HttpServletRequest request, HttpServletResponse response, Context cx, Scriptable scope) {
         super(scope, ScriptableObject.getObjectPrototype(scope));
 
-        Reference ref = request.getResourceRef();
-
-        this.put("method", this, request.getMethod().toString());
-
-        this.put("scriptName", this, ref.getLastSegment());
-
-        List<String> seg = new ArrayList<String>(ref.getSegments().subList(4, ref.getSegments().size()));
-        seg.add(0, "");
-        this.put("pathInfo", this, StringUtils.join(seg.toArray(), "/"));
-        
-        this.put("queryString", this, ref.getQuery());
-        
-        this.put("host", this, ref.getHostDomain());
-        
-        this.put("port", this, ref.getHostPort());
-        
-        this.put("scheme", this, ref.getScheme());
-
-        try {
-            // TODO: deal with input
-            this.put("input", this, Context.javaToJS(request.getEntity().getStream(), scope));
-        } catch (IOException e) {
-            LOGGER.warning(e.getMessage());
-        }
-
-        Scriptable headers = cx.newObject(scope);
-        Form requestHeaders = (Form) request.getAttributes().get("org.restlet.http.headers");
-        for (String name : requestHeaders.getNames()) {
-            String value = requestHeaders.getFirstValue(name, true);
-            headers.put(name.toLowerCase(), headers, value);
-        }
-        this.put("headers", this, headers);
+//        Reference ref = request.getResourceRef();
+//
+//        this.put("method", this, request.getMethod().toString());
+//
+//        this.put("scriptName", this, ref.getLastSegment());
+//
+//        List<String> seg = new ArrayList<String>(ref.getSegments().subList(4, ref.getSegments().size()));
+//        seg.add(0, "");
+//        this.put("pathInfo", this, StringUtils.join(seg.toArray(), "/"));
+//
+//        this.put("queryString", this, ref.getQuery());
+//
+//        this.put("host", this, ref.getHostDomain());
+//
+//        this.put("port", this, ref.getHostPort());
+//
+//        this.put("scheme", this, ref.getScheme());
+
+//        try {
+//            // TODO: deal with input
+//            this.put("input", this, Context.javaToJS(request.getEntity().getStream(), scope));
+//        } catch (IOException e) {
+//            LOGGER.warning(e.getMessage());
+//        }
+
+//        Scriptable headers = cx.newObject(scope);
+//        Form requestHeaders = (Form) request.getAttributes().get("org.restlet.http.headers");
+//        for (String name : requestHeaders.getNames()) {
+//            String value = requestHeaders.getFirstValue(name, true);
+//            headers.put(name.toLowerCase(), headers, value);
+//        }
+//        this.put("headers", this, headers);
```

Comment:
Why is this commented out? If it is unnecessary, remove it.

Reply:
fixed.